packages/core-data/src/selectors.ts

```diff
@@ -936,6 +936,9 @@ export function hasRedo( state: State ): boolean {
  * @return The current theme.
  */
 export function getCurrentTheme( state: State ): any {
+	if ( ! state.currentTheme ) {
+		return null;
+	}
```

Comment:
This is a fix that was forced by the "`null` state instead of `undefined`" changes, and it can be merged even though the `state.currentTheme` remains `undefined`. `getEntityRecord` should never get a fourth argument (`key`) that's not a valid string, its type doesn't allow that. The `State` type for the `getCurrentTheme`'s `state` argument "lies" about the type of `state.currentTheme`. It can be `undefined` (or `null`) when there is no active theme or when the active theme is not yet fetched from the REST endpoint.

Reply:
I realize this is pre-existing, but there are places in the codebase that rely on the `getCurrentTheme()` to always return a non-nullish value, for example:

https://github.com/WordPress/gutenberg/blob/dd7e10adb416efeb213b79ffb5a97a23e454fc1e/packages/block-library/src/pattern/edit.js#L23

Should we be concerned that those potentially might cause issues under some conditions?

Reply:
If `state.currentTheme` is not a valid theme slug, then `getEntityRecord` would return `undefined` anyway, because the state never has a theme whose slug (ID) would be `undefined` or `null`. So, it's not that we're now returning `null` where a theme object would be previously returned.

The existing places in the codebase are incorrect, because `getCurrentTheme()` is not guaranteed to return a valid theme. The code relies on the fact that by the time it runs, the `/wp/v2/themes?status=active` REST request has already finished loading.

Reply:
Agreed - the existing code would throw an error before with the `undefined`, and will throw an error now with the `null`. 

IMHO these cases can be handled in another PR, and are something that an eventual TS migration will catch anyway.